homeassistant/components/alert.py

```diff
@@ -0,0 +1,233 @@
+
+
+def toggle(hass, entity_id):
+    """Toggle acknowledgement of alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.services.call(DOMAIN, SERVICE_TOGGLE, data)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup alert component."""
+    alerts = config.get(DOMAIN)
+
+    @asyncio.coroutine
+    def async_handle_alert_service(service_call):
+        """Handle calls to alert services."""
+        alert_ids = service.extract_entity_ids(hass, service_call)
+
+        for alert_id in alert_ids:
+            alert = ALL_ALERTS[alert_id]
+            if service_call.service == SERVICE_TURN_ON:
+                yield from alert.async_turn_on()
+            elif service_call.service == SERVICE_TOGGLE:
+                yield from alert.async_toggle()
+            else:
+                yield from alert.async_turn_off()
+
+    for alert in alerts.get(CONF_ENTITIES):
+        entity = yield from hass.loop.run_in_executor(
+            None, Alert, hass, alert[CONF_NAME], alert[CONF_ENTITY_ID],
+            alert.get(CONF_STATE, STATE_ON), alert[CONF_REPEAT],
+            alert.get(CONF_SKIP_FIRST, False),
+            alert[CONF_NOTIFIERS])
+        ALL_ALERTS[entity.entity_id] = entity
+
+    descriptions = {}
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_TURN_OFF, async_handle_alert_service,
+        descriptions.get(SERVICE_TURN_OFF), schema=ALERT_SERVICE_SCHEMA)
+    hass.services.async_register(
+        DOMAIN, SERVICE_TURN_ON, async_handle_alert_service,
+        descriptions.get(SERVICE_TURN_ON), schema=ALERT_SERVICE_SCHEMA)
+    hass.services.async_register(
+        DOMAIN, SERVICE_TOGGLE, async_handle_alert_service,
+        descriptions.get(SERVICE_TOGGLE), schema=ALERT_SERVICE_SCHEMA)
+
+    return True
+
+
+class Alert(ToggleEntity):
+    """Representation of an alert."""
+
+    def __init__(self, hass, name, entity, state, repeat, skip_first,
+                 notifiers):
+        """Initialize the alert."""
+        self.hass = hass
+        self._name = name
+        self._alert_state = state
+        self._repeat = timedelta(minutes=repeat)
+        self._skip_first = skip_first
+        self._notifiers = notifiers
+        self._firing = False
+        self._ack = False
+        self._cancel = None
+        self.entity_id = DOMAIN + '.%s' % slugify(name)
+
+        event.async_track_state_change(hass, entity,
+                                       self.watched_entity_change)
+
+        self.update_ha_state()
+
+    @property
+    def name(self):
+        """Return the name of the URL."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the URL."""
+        if self._firing:
+            if self._ack:
+                return STATE_OFF
+            return STATE_ON
+        return STATE_IDLE
+
+    @property
+    def hidden(self):
+        """Hide the alert when it is not firing."""
+        return not self._firing
```

Comment:
Wait... I think I just figured out your question. My bad.

This is an alarm, it cannot be silenced if it is not firing. Because there is nothing to silence. It is already silent.